cupyx/scipy/signal/signaltools.py

```diff
@@ -0,0 +1,81 @@
+import math
+
+import cupy
+
+
+def _numeric_arrays(arrays, kinds='buifc'):
+    if type(arrays) == cupy.ndarray:
+        return arrays.dtype.kind in kinds
+    for array in arrays:
+        if array.dtype.kind not in kinds:
+            return False
+    return True
+
+
+def choose_conv_method(in1, in2, mode='full'):
+    """Find the fastest convolution/correlation method.
+
+    Args:
+        in1 (cupy.ndarray): first input.
+        in2 (cupy.ndarray): second input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        str: A string indicating which convolution method is fastest,
+         either ‘direct’ or ‘fft’.
+
+    .. warning::
+        This function currently doesn't support measure option,
+        nor multidimensional inputs.
+
+    .. seealso:: :func:`scipy.signal.choose_conv_method`
+
+    """
+    if in1.ndim != 1 or in2.ndim != 1:
+        raise NotImplementedError('Only 1d inputs are supported currently')
+
+    if _numeric_arrays([in1, in2], kinds='bui'):
+        return 'direct'
+
+    if _fftconv_faster(in1, in2, mode):
+        return 'fft'
+
+    return 'direct'
+
+
+def _fftconv_faster(x, h, mode):
+    """
+    .. seealso:: :func: `scipy.signal.signaltools._fftconv_faster`
+
+    """
+    fft_ops, direct_ops = _conv_ops(x.size, h.size, mode)
+    offset = -1e-3
+    constants = {
```

Comment:
Please explain what this constants are. I'm wondering these constants are useful also for GPU computations.

Reply:
These constants for SciPy were determined using code from https://github.com/stsievert/scipy-convolve-timings as discussed at:  https://github.com/scipy/scipy/pull/11031#issuecomment-552068555

I expect running the same sort of thing on the GPU might give substantially different results. I am not sure how feasible it is to get a single set of numbers that applies in general across a wide range of GPUs, but a single set based on a given GPU would probably still be a better match than these existing CPU-based constants.